src/reactor.h

```diff
@@ -0,0 +1,342 @@
+
+  //WARNING: The default on this is completely arbitrary!
+  #pragma cyclus var { \
+    "default": 0.05, \
+    "doc": "Percent of blanket that gets recycled every blanket turnover period", \
+    "tooltip": "Defaults to 0.05 (5%), must be between 0 and 15%", \
+    "units": "dimensionless", \
+    "uitype": "range", \
+    "range": [0, 0.15], \
+    "uilabel": "Blanket Turnover Rate" \
+  }
+  double blanket_turnover_rate;
+
+  #pragma cyclus var { \
+    "default": 1, \
+    "doc": "number of timesteps between blanket recycles", \
+    "tooltip": "Defaults to 0.05 (5%), must be between 0 and 15%", \
+    "units": "dimensionless", \
+    "uitype": "range", \
+    "range": [0, 1000], \
+    "uilabel": "Blanket Turnover Rate" \
+  }
+  int blanket_turnover_frequency;
+
+  #pragma cyclus var {"tooltip":"Buffer for handling tritium material to be used in reactor"}
+  cyclus::toolkit::ResBuf<cyclus::Material> tritium_storage;
+
+  #pragma cyclus var {"tooltip":"Buffer for handling excess tritium material to be sold"}
+  cyclus::toolkit::ResBuf<cyclus::Material> tritium_excess;
+
+  #pragma cyclus var {"tooltip":"Buffer for handling tritium sequestered in the system"}
+  cyclus::toolkit::ResBuf<cyclus::Material> tritium_sequestered;
+
+  #pragma cyclus var {"tooltip":"Buffer for handling helium-3 byproduct material"}
+  cyclus::toolkit::ResBuf<cyclus::Material> helium_storage;
+
+  #pragma cyclus var {"tooltip":"Buffer for handling enriched lithium blanket material"}
+  cyclus::toolkit::ResBuf<cyclus::Material> blanket;
+
+  #pragma cyclus var {"tooltip":"Tracker to handle on-hand tritium"}
+  cyclus::toolkit::TotalInvTracker fuel_tracker;
+
+  #pragma cyclus var {"tooltip":"Tracker to handle blanket material"}
+  cyclus::toolkit::TotalInvTracker blanket_tracker;
+
+  cyclus::toolkit::MatlBuyPolicy fuel_startup_policy;
+  cyclus::toolkit::MatlBuyPolicy fuel_refill_policy;
+
+  cyclus::toolkit::MatlBuyPolicy blanket_fill_policy;
+
+  cyclus::toolkit::MatlSellPolicy tritium_sell_policy;
+  cyclus::toolkit::MatlSellPolicy helium_sell_policy;
+
+  void Startup();
+  void OperateReactor(double TBR);
+  void SequesterTritium();
+  void DecayInventory(cyclus::toolkit::ResBuf<cyclus::Material> &inventory);
+  void CombineInventory(cyclus::toolkit::ResBuf<cyclus::Material> &inventory);
+  void ExtractHelium(cyclus::toolkit::ResBuf<cyclus::Material> &inventory);
+  void RecordEvent(std::string name, std::string val);
+  void RecordStatus(std::string Status, double power);
+  void RecordInventories(double storage, double excess, double sequestered, double blanket, double helium);
+  void RecordOperationalInfo(std::string name, std::string val);
+  void DepleteBlanket(double bred_tritium_mass);
+  cyclus::Material::Ptr BreedTritium(double fuel_usage, double TBR);
+  std::string GetComp(cyclus::Material::Ptr mat);
+
+
+  private: 
+    bool sufficient_tritium_for_operation = false;
+    const double seconds_per_year = 31536000.0;
```

Comment:
Isn't this defined at the cyclus level?

Reply:
The number of seconds in a month is defined as the number of seconds in a year/12 I think, so I could use that*12 instead, but I checked the value and it's not quite 1/12 of a year (although how long a year is depends on who you ask really, NASA says 365.2422, some people say 365.25, some people say 365, none of those lead to a value which matches what's in cyclus which is 365.25639 days, or what appears to be a gaussian year, not totally sure since it doesn't match exactly, but maybe that's a "we wanted it to be an int" rounding thing). I do not think cyclus defines the seconds in a year explicitly, at least not in context. Not sure the paragraph was totally necessary, I'll change this to context()->kDefaultTimeStepDur*12 for consistency.

Reply:
I don't think I can access that variable, actually, since it never actually makes its way into context, and is just defined in the header and used as a default. I can't just use dt() since it's possible to change from the default, too. Just going to make it the same number times 12 for consistency. If I'm missing something, though, feel free to let me know.